EIPS/eip-7923.md

```diff
@@ -169,6 +176,46 @@ index bf34fbee..f63b7460 100644
 
      def memory_write(self, start_position: int, size: int, value: bytes) -> None:
          return self._memory.write(start_position, size, value)
+diff --git a/eth/vm/forks/frontier/computation.py b/eth/vm/forks/frontier/computation.py
+index 51666ae0..443f82b5 100644
+--- a/eth/vm/forks/frontier/computation.py
++++ b/eth/vm/forks/frontier/computation.py
+@@ -29,6 +29,7 @@ from eth.exceptions import (
+     InsufficientFunds,
+     OutOfGas,
+     StackDepthLimit,
++    VMError,
+ )
+ from eth.vm.computation import (
+     BaseComputation,
+@@ -87,12 +88,21 @@ class FrontierComputation(BaseComputation):
+
+         state.touch_account(message.storage_address)
+
+-        computation = cls.apply_computation(
+-            state,
+-            message,
+-            transaction_context,
+-            parent_computation=parent_computation,
+-        )
++        num_pages_anchor = transaction_context.num_pages
++        try:
++            computation = cls.apply_computation(
++                state,
++                message,
++                transaction_context,
++                parent_computation=parent_computation,
++            )
++        finally:
++            # "deallocate" all the pages allocated in the child computation
++
++            # sanity check an invariant:
++            allocated_pages = len(computation._memory.pages)
++            assert transaction_context.num_pages = num_pages_anchor + allocated pages
```

Comment:
Shouldn't this be `==`?

Reply:
This assert will not work if we throw the out of memory, because it will add `1` to `num_pages` but it will not add it to the `self.memory._pages`. So it will be off by one in case of an OOM and this assert will fail.

Reply:
> This assert will not work if we throw the out of memory, because it will add 1 to num_pages but it will not add it to the self.memory._pages. So it will be off by one in case of an OOM and this assert will fail.

that was a wicked nice catch! i think d951ba52246f5c43ebc402ebf8a4b7538ecafac2 fixes

Reply:
and 90d8b589662c8b753af06b18c9da2c5997b9ebed for the typo